internal/rbuffer/ring_buffer.go

```diff
@@ -96,6 +96,7 @@ func (b *RingBuffer) read(p []byte) (n int, err error) {
 		copy(p[c1:], b.buf[0:c2])
 	}
 	b.r = (b.r + n) % b.size
+	b.isFull = false
```

Comment:
Do we need a check here to make sure that `n == b.size`? @adambabik

Reply:
Yes, that's the root cause I also identified. I think there is one more place to change: https://github.com/stateful/runme/pull/151.